packages/matter-node.js/src/persistence/StorageNodeLocalstorage.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright 2022-2023 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { LocalStorage } from "node-localstorage";
+import { Storage } from "./Storage";
+import { fromJson, SupportedStorageTypes, toJson } from "./JsonConverter";
+
+export class StorageNodeLocalstorage implements Storage {
+    private readonly localStorage;
+
+    constructor(
+        path: string,
+    ) {
+        this.localStorage = new LocalStorage(path);
+    }
+
+    async initialize() {
+        // nothing to do
+    }
+
+    async close() {
+        // nothing to do
+    }
+
+    buildStorageKey(context: string, key: string): string {
+        return context + "---" + key;
```

Comment:
I'd make the context separator simply `:`.

Reply:
Is a colon a good idea because used in Filenames? I tried to use a char which do not make issues there and is still "unique" enough that no one use it. I also thought about "." And forbidding . In context. What about that?

Reply:
I adjusted based on my proposal with "." (yes the ":" should be handled by node-localstorage but .... ;-)

Reply:
Ok.  Fair enough.  `-` and `_` and `.` are far more standard than `:`.